x/denommetadata/keeper/msg_server.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/dymensionxyz/dymension-rdk/x/denommetadata/types"
+)
+
+var _ types.MsgServer = &Keeper{}
+
+// CreateDenomMetadata create the denom metadata in bank module
+func (k Keeper) CreateDenomMetadata(
+	goCtx context.Context,
+	msg *types.MsgCreateDenomMetadata,
+) (*types.MsgCreateDenomMetadataResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
+
+	if !k.IsAddressPermissioned(ctx, msg.SenderAddress) {
+		return nil, types.ErrNoPermission
+	}
+
+	found := k.bankKeeper.HasDenomMetaData(ctx, msg.TokenMetadata.Base)
+	if found {
+		return nil, types.ErrDenomAlreadyExists
+	}
+
+	k.bankKeeper.SetDenomMetaData(ctx, msg.TokenMetadata)
+	return &types.MsgCreateDenomMetadataResponse{}, nil
+}
+
+// UpdateDenomMetadata update the denom metadata in bank module
+func (k Keeper) UpdateDenomMetadata(
```

Comment:
I might be missing the point, but do we need to have both Create and Update handlers? They do the same thing anyway. Why not just have a SetDenomMetadata handler?

Reply:
I implemented it based on the requirement that @omritoptix posted in the original issue. My idea is that you want two function so that when you register, you don't accidentally setup the same thing twice (make sure that you intended); with update, you won't accidentally create new denommetadata because of input.